tests/test_mcp.py

```diff
@@ -127,6 +134,35 @@ async def test_agent_with_stdio_server(allow_model_requests: None, agent: Agent)
         )
 
 
+async def test_agent_with_conflict_tool_name(agent: Agent):
+    @agent.tool_plain
+    def get_none() -> None:
+        """Return nothing"""
+        return None
+
+    async with agent.run_mcp_servers():
+        with pytest.raises(UserError, match='tool name conflicts with existing tool'):
+            await agent.run('Get me a conflict')
+
+
+async def test_agent_with_prefix_tool_name(openai_api_key: str):
+    server = MCPServerStdio('python', ['-m', 'tests.mcp_server'], tool_prefix='foo')
+    model = OpenAIModel('gpt-4o', provider=OpenAIProvider(api_key=openai_api_key))
+    agent = Agent(
+        model,
+        mcp_servers=[server],
+    )
+
+    @agent.tool_plain
+    def get_none() -> None:
+        """Return nothing"""
+        return None
+
+    async with agent.run_mcp_servers():
+        with pytest.raises(RuntimeError, match='Model requests are not allowed, since ALLOW_MODEL_REQUESTS is False'):
```

Comment:
This means that we passed the prepare reques and there is no conflict in the tool name

Reply:
If there's no better way to do this (ensure it doesn't raise), can we add a comment to this effect because it looks confusing?